src/lib.rs

```diff
@@ -371,10 +376,16 @@ pub mod pallet {
 			let sindex = Self::get_supersig_index_from_id(&supersig_id)
 				.ok_or(Error::<T>::SupersigNotFound)?;
 			let mut new_members = new_members;
+			// cannot fail
+			let old_members = Self::supersigs(sindex).unwrap().members;
+			new_members.retain(|memb| !old_members.contains(memb));
+			let deposit = <BalanceOf<T>>::from(size_of::<T::AccountId>() as u32)
+				.saturating_mul((new_members.len() as u32).into())
+				.saturating_mul(T::PricePerBytes::get());
+			T::Currency::reserve(&supersig_id, deposit)?;
 
 			Supersigs::<T>::mutate(sindex, |wrapped_supersig| {
 				if let Some(supersig) = wrapped_supersig {
-					new_members.retain(|memb| !supersig.members.contains(memb));
 					for _ in 0..new_members.len() {
 						// this will never fail, because we set provider to 1 when creating the
 						// supersig
```

Comment:
Not true. Can hit `MaxConsumers::max_consumers()` and fail. We should handle the error

Reply:
And we can get rid of the whole inc_consumer here. It does not make sense. There is not one more runtime pallet depending on the supersig for each member

Reply:
Also just bellow:
`					supersig.members.append(new_members.as_mut());`
append leave the vector you pass in empty. That's why it have to be mutable.
So your `UsersAdded` event will receive an empty vector as argument. Not what we want